### DISCUSSION TITLE
Validation on array field

### DISCUSSION THREAD

**Discussion by dawidmachon** [payloadcms/payload#2065] [Q&A]
Hi,
imagine an array field, which keeps different users.
```js
{
      name: "players",
      label: "Players",
      type: "array",
      fields: [
        {
          name: "user",
          label: "User",
          type: "relationship",
          relationTo: "users",
          hasMany: false,
          required: true,
        },
      validate: ()=>{}
}
```

How would you write or just give an idea about: Checking if user already is in array?
Problem -> when you are creating document with array field, then all entries are there at once, without _id's, so what only comes to my mind is going each by each and compare against all - but this is super slow and not performant.

Maybe someone has better approach to such topic. **How validate on array fields to enforce unique entries inside?**

**Comment by DanRibbens** [comment]
Have you looked at using the `filterOptions` on your relationship field?

You can use it to limit the options available in the UI and it also enforces this in your API in the built-in validation of the relationship field. This way you won't have a `validate` function at all.

Your `filterOptions` will be a function that uses the incoming data and returns `{ not_in: userIDs }` where userIDs is an array of user IDs excluding the assigned value on the field (which you also get as an argument).

Let me know if that helps!

**Reply by dawidmachon** [reply]
Thanks for good tip. It is much better approach I think. 
But it is not solving my issue still. Because in `filterOptions` you mentioned: "I will get access to the value of field which is currently 'in use'." But that's not true from my perspective. 

There are only these options:
<html>
<body>


key | desc
-- | --
relationTo | The relationTo to filter against (as defined on the field)
data | An object of the full collection or global document currently being edited
siblingData | An object of the document data limited to fields within the same parent to the field
id | The value of the collection id, will be undefined on create request
user | The currently authenticated user object


</body>
</html>

So when I'm posting create document from API only on validate function, I got access to current value on field. Or am I missing something?

On top of that - on validate function, I also have issues. Because `data` always has all fields in array, besides for which iteration value we are currently validating. (which is okay, I suppose). But sibling data contains only 'recent' iteration. So, the only way which right now I'm thinking is checking iv current value exists in `data` more than once.
